search-commons/src/main/java/no/unit/nva/search/SearchClient.java

```diff
@@ -95,7 +97,12 @@ private BoolQueryBuilder matchOneOfOrganizationIdsQuery(ViewingScope viewingScop
         for (URI excludedOrganizationId : viewingScope.getExcludedUnits()) {
             queryBuilder.mustNot(matchPhraseQuery(ORGANIZATION_IDS, excludedOrganizationId.toString()));
         }
-        queryBuilder.mustNot(QueryBuilders.matchQuery(STATUS, APPROVED));
+        searchQueryToRemoveApprovedWorkListAndDraftPublications(queryBuilder);
         return queryBuilder;
     }
+
+    private void searchQueryToRemoveApprovedWorkListAndDraftPublications(BoolQueryBuilder queryBuilder) {
+        queryBuilder.mustNot(QueryBuilders.matchQuery(STATUS, APPROVED));
+        queryBuilder.mustNot(QueryBuilders.matchQuery(PUBLICATION_STATUS, DRAFT));
```

Comment:
is it not possible to say that we do not want PublicationStatus.DRADT only for DoiRequests?

Reply:
BoolQueryBuilder have must, mustNot, should and filter clauses, I am not able to fine clauses to match our scenario like if it's DoiRequest Type then check for Publication Status else return.

Reply:
Have you tried combining multiple boolean queries? https://stackoverflow.com/questions/28538760/elasticsearch-bool-query-combine-must-with-or

Reply:
Tried with different combinations of boolean queries but not getting the exact result. For the time being will forward with current implementation ie. Look for publication.status = DRAFT.

Will create one bug task in Jira to fix this in future. And the fix needed : Check for publication.status only if it is a DoiRequest Type

Reply:
As suggested created one bug in Jira

https://unit.atlassian.net/browse/NP-9090